sympy/physics/continuum_mechanics/beam.py

```diff
@@ -491,6 +529,171 @@ def apply_support(self, loc, type="fixed"):
+
+        Examples
+        ========
+        There is a beam of length 15 meters. Pin supports are placed at distances
+        of 0 and 10 meters. There is a fixed support at the end. There are two rotation hinges
+        in the structure, one at 5 meters and one at 10 meters. A pointload of magnitude
+        10 kN is applied on the hinge at 5 meters. A distributed load of 5 kN works on
+        the structure from 10 meters to the end.
+
+        Using the sign convention of upward forces and clockwise moment
+        being positive.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(15, E, I)
+        >>> r0 = b.apply_support(0, type='pin')
+        >>> r10 = b.apply_support(10, type='pin')
+        >>> r15, m15 = b.apply_support(15, type='fixed')
+        >>> p5 = b.apply_rotation_hinge(5)
+        >>> p12 = b.apply_rotation_hinge(12)
+        >>> b.apply_load(-10, 5, -1)
+        >>> b.apply_load(-5, 10, 0, 15)
+        >>> b.solve_for_reaction_loads(r0, r10, r15, m15)
+        >>> b.reaction_loads
+        {M_15: -75/2, R_0: 0, R_10: 40, R_15: -5}
+        >>> b.rotation_jumps
+        {P_12: -1875/(16*E*I), P_5: 9625/(24*E*I)}
+        >>> b.rotation_jumps[p12]
+        -1875/(16*E*I)
+        >>> b.bending_moment()
+        -9625*SingularityFunction(x, 5, -1)/24 + 10*SingularityFunction(x, 5, 1)
+        - 40*SingularityFunction(x, 10, 1) + 5*SingularityFunction(x, 10, 2)/2
+        + 1875*SingularityFunction(x, 12, -1)/16 + 75*SingularityFunction(x, 15, 0)/2
+        + 5*SingularityFunction(x, 15, 1) - 5*SingularityFunction(x, 15, 2)/2
+        """
+        loc = sympify(loc)
+        E = self.elastic_modulus
+        I = self._get_I(loc)
+
+        #Check for duplicate hinges
+        if loc in self._applied_rotation_hinges:
+            raise ValueError('Cannot place two rotation hinges at the same location.')
+
+        #Check for unusual hinge placement
+        warning_message = 'It is highly unusual to place a rotation hinge and a sliding hinge at the same location.'
+        if loc in self._applied_sliding_hinges:
+            warnings.warn(warning_message)
+
+        #Check for other errors
+        if loc == 0:
+            raise ValueError('Cannot place hinge at the beginning of the beam.')
+        if loc == self.length:
+            raise ValueError('Cannot place hinge at the end of the beam.')
+        if any(loc == arg[1] and arg[2] == -2 for arg in self._applied_loads):
+            raise ValueError('Cannot place rotation hinge at the location of a moment load.')
+        if any(loc == support[0] and support[1] == 'fixed' for support in self._applied_supports):
+            raise ValueError('Cannot place rotation hinge at the location of a fixed support. Change fixed support to pin.')
+
+        rotation_jump = Symbol('P_'+str(loc))
```

Comment:
This works nicely if `loc` is an integer, but what about if it is a float or a symbolic expression?

Reply:
This should not be a problem. This is symbol is returned and the user can give it any other name they want. For example in this code:
```
b = Beam(13, 20, 20)
r0, m0 = b.apply_support(0, type="fixed")
w333 = b.apply_sliding_hinge(10/3)
r13 = b.apply_support(13, type="pin")
b.apply_load(-10, 5, -1)
b.solve_for_reaction_loads(r0, m0, r13)
print(b.deflection_jumps)
print(b.deflection_jumps[w333])
```
The output is: 
`{W_3.33333333333333: -443/30}`
`-443/30`

The string of the location is there so the user can know which hinge is meant when there are multiple hinges in the beam. The value op the jump can be called by using the name they gave to the symbol of the jump. This works the same way when a symbolic expression is used to describe the location of the hinge. 

```
b = Beam(L, 20, 20)
r0, m0 = b.apply_support(0, type="fixed")
wL3 = b.apply_sliding_hinge(L/3)
rL = b.apply_support(L, type="pin")
b.apply_load(-10, L/2, -1)
b.solve_for_reaction_loads(r0, m0, rL)
print(b.deflection_jumps)
print(b.deflection_jumps[wL3])
```
Output: 
`{W_L/3: -11*L**3/1920}`
`-11*L**3/1920`

Reply:
Yes, symbols can be named like this, but such symbol names can break code printers and may have other undesired effects when used in other parts of sympy. Most other functions that need to create symbol names for the user keep them simple (like `C1,C2,...` in `dsolve()`). I see that we are already doing this kind of string creation:

```
grep -r Symbol sympy/physics/continuum_mechanics/beam.py 
from sympy.core import S, Symbol, diff, symbols
    def __init__(self, length, elastic_modulus, second_moment, area=Symbol('A'), variable=Symbol('x'), base_char='C'):
            A Symbol or value representing the Beam's length.
        area : Symbol/float
        variable : Symbol, optional
            A Symbol object that will be used as the variable along the beam
            curve. By default, it is set to ``Symbol('x')``.
        to ``Symbol('x')``, but this property is mutable.
        if isinstance(v, Symbol):
            raise TypeError("""The variable should be a Symbol object.""")
        Symbol or tuple of Symbol
            - Symbol(reaction_force) if type = "pin" or "roller"
            - Symbol(reaction_force), Symbol(reaction_moment) if type = "fixed"
            reaction_load = Symbol('R_'+str(loc))
            reaction_load = Symbol('R_'+str(loc))
            reaction_moment = Symbol('M_'+str(loc))
        h = Symbol('h')     # Force due to hinge
        C1 = Symbol('C1')
        C2 = Symbol('C2')
        C3 = Symbol('C3')
        C4 = Symbol('C4')
        C3 = Symbol('C3')
        C4 = Symbol('C4')
        C3 = Symbol('C3')
        C4 = Symbol('C4')
            Python dictionary containing Symbols as key and their
        for sym in shear_stress.atoms(Symbol):
            Python dictionary containing Symbols as key and their
        for sym in shear_force.atoms(Symbol):
            Python dictionary containing Symbols as key and their
        for sym in bending_moment.atoms(Symbol):
            Python dictionary containing Symbols as key and their
        for sym in slope.atoms(Symbol):
            Python dictionary containing Symbols as key and their
        for sym in deflection.atoms(Symbol):
               Python dictionary containing Symbols as key and their
        for sym in self.deflection().atoms(Symbol):
        C3 = Symbol('C3')
        C4 = Symbol('C4')
               Python dictionary containing Symbols as key and their
            for sym in self._ild_reactions[reaction].atoms(Symbol):
        for sym in self._length.atoms(Symbol):
               Python dictionary containing Symbols as key and their
        for sym in self._ild_shear.atoms(Symbol):
        for sym in self._length.atoms(Symbol):
               Python dictionary containing Symbols as key and their
        for sym in self._ild_moment.atoms(Symbol):
        for sym in self._length.atoms(Symbol):
        # checking whether length is an expression in terms of any Symbol.
            l = list(self.length.atoms(Symbol))
            # assigning every Symbol a default value of 10
                 area, variable=Symbol('x')):
            A Symbol or value representing the Beam's length.
        variable : Symbol, optional
            A Symbol object that will be used as the variable along the beam
            curve. By default, it is set to ``Symbol('x')``.
            reaction_load = Symbol('R_'+str(loc))
            reaction_load = Symbol('R_'+str(loc))
            reaction_moment = Symbol('M_'+str(loc))
        C1 = Symbol('C1')
        C2 = Symbol('C2')
        C_i = Symbol('C_i')
        for sym in shear_force[dir_num].atoms(Symbol):
            Python dictionary containing Symbols as key and their
        for sym in bending_moment[dir_num].atoms(Symbol):
            Python dictionary containing Symbols as key and their
        for sym in slope[dir_num].atoms(Symbol):
            Python dictionary containing Symbols as keys and their
        for sym in deflection[dir_num].atoms(Symbol):
            Python dictionary containing Symbols as keys and their
               Python dictionary containing Symbols as key and their
        for sym in shear_stress[dir_num].atoms(Symbol):
            Python dictionary containing Symbols as key and their
```

So, it is fine to continue to do so.